pkg/v1/providers/yttcb/clusterbootstrap.yaml

```diff
@@ -0,0 +1,218 @@
+#@ if data.values.VSPHERE_CONTROL_PLANE_ENDPOINT:
+#@   vsphereControlPlaneEndpoint, err = assert.try_to(lambda : ip.parse_addr(data.values.VSPHERE_CONTROL_PLANE_ENDPOINT))
+#@   if err == None:
+#@     if vsphereControlPlaneEndpoint.is_ipv4():
+#@       vsphereControlPlaneEndpointAsCidr = vsphereControlPlaneEndpoint.string() + "/32"
+#@     end
+#@     if vsphereControlPlaneEndpoint.is_ipv6():
+#@       vsphereControlPlaneEndpointAsCidr = vsphereControlPlaneEndpoint.string() + "/128"
+#@     end
+#@   end
+#@ end
+
+#@ def antrea_configs_exist():
+#@   return (data.values.NSXT_POD_ROUTING_ENABLED or
+#@           data.values.ANTREA_NO_SNAT or
+#@           data.values.ANTREA_TRAFFIC_ENCAP_MODE != "encap" or
+#@           not data.values.ANTREA_PROXY or
+#@           not data.values.ANTREA_ENDPOINTSLICE or
+#@           not data.values.ANTREA_POLICY or
+#@           data.values.ANTREA_NODEPORTLOCAL or
+#@           not data.values.ANTREA_TRACEFLOW or
+#@           data.values.ANTREA_EGRESS or
+#@           data.values.ANTREA_FLOWEXPORTER or
+#@           data.values.ANTREA_DISABLE_UDP_TUNNEL_OFFLOAD)
+#@ end
+
+#@ def vspherecpi_configs_exist():
+#@   return data.values.VSPHERE_SERVER or data.values.NSXT_MANAGER_HOST
+#@ end
+
+#@ def vspherecsi_configs_exist():
+#@   return (not data.values.VSPHERE_INSECURE or
+#@           data.values.USE_TOPOLOGY_CATEGORIES)
+#@ end
+
+#@ def should_create_clusterbootstrap():
+#@   return data.values.CNI != "antrea" or antrea_configs_exist() or vspherecsi_configs_exist() or vspherecpi_configs_exist()
+#@ end
+
+#@ if should_create_clusterbootstrap():
+---
+apiVersion: run.tanzu.vmware.com/v1alpha3
+kind: ClusterBootstrap
+metadata:
+  name: #@ data.values.CLUSTER_NAME
+  namespace: #@ data.values.NAMESPACE
+  annotations:
+    tkg.tanzu.vmware.com/add-missing-fields-from-tkr: #@ data.values.KUBERNETES_RELEASE
+spec:
+  cni:
+    #@ if data.values.CNI == "antrea":
+    refName: antrea*
+    valuesFrom:
+      providerRef:
+        apiGroup: cni.tanzu.vmware.com
+        kind: AntreaConfig
+        name: #@ data.values.CLUSTER_NAME
+    #@ elif data.values.CNI == "calico":
+    refName: calico*
+    valuesFrom:
+      providerRef:
+        apiGroup: cni.tanzu.vmware.com
+        kind: CalicoConfig
+        name: #@ data.values.CLUSTER_NAME
+    #@ end
+  #@ if vspherecpi_configs_exist() and data.values.PROVIDER_TYPE == "vsphere" and not data.values.IS_WINDOWS_WORKLOAD_CLUSTER:
+  cpi:
+    refName: vsphere-cpi*
+    valuesFrom:
+      providerRef:
+        apiGroup: cpi.tanzu.vmware.com
+        kind: VSphereCPIConfig
+        name: #@ data.values.CLUSTER_NAME
+  #@ end
+  #@ if vspherecsi_configs_exist() and data.values.PROVIDER_TYPE == "vsphere" and not data.values.IS_WINDOWS_WORKLOAD_CLUSTER:
+  csi:
+    refName: vsphere-csi*
+    valuesFrom:
+      providerRef:
+        apiGroup: csi.tanzu.vmware.com
+        kind: VSphereCSIConfig
+        name: #@ data.values.CLUSTER_NAME
+  #@ end
+  additionalPackages:
```

Comment:
@ankeesler , is pinniped package needed in list of additional packages here.
cc: @vijaykatam